pkg/subctl/cmd/validate_k8sconfig.go

```diff
@@ -0,0 +1,76 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+)
+
+var supportedNetworkPlugins = []string{"generic", "canal-flannel", "weave-net", "OpenShiftSDN", "OVNKubernetes"}
+
+var validateK8sCmd = &cobra.Command{
+	Use:   "cluster-config",
+	Short: "Validate the Kubernetes cluster configuration",
+	Long: "This command validates if the Kubernetes cluster configuration is supported by Submariner," +
+		" including version, CNI, and kube-proxy mode.",
+	Run: validateK8sConfig,
+}
+
+func init() {
+	validateCmd.AddCommand(validateK8sCmd)
+}
+
+func validateK8sConfig(cmd *cobra.Command, args []string) {
+	configs, err := getMultipleRestConfigs(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config for cluster", err)
+
+	for _, item := range configs {
+		message := fmt.Sprintf("Validating Kubernetes cluster configuration in %q", item.clusterName)
+		status.Start(message)
+		fmt.Println()
+		submariner := getSubmarinerResource(item.config)
+
+		if submariner == nil {
+			fmt.Println(submMissingMessage)
```

Comment:
In case Submariner is not installed, the user will not be notified on the current cluster the check had failed

Reply:
We will get a print message, but it does not print the cluster name and show it as a failure. I hope you meant the same. Let me update the code to use status.QueueFailureMessage

Thanks @maayanf24